storage/interceptors/broker_notifications_interceptor.go

```diff
@@ -164,19 +158,79 @@ func fetchBrokerPlans(ctx context.Context, brokerID string, repository storage.R
 	return objList.(*types.ServicePlans).ServicePlans, nil
 }
 
-func getSupportedPlatformsForPlans(plans []*types.ServicePlan) []string {
-	platformTypes := make(map[string]bool)
+func ResolveSupportedPlatformIDsForPlans(ctx context.Context, plans []*types.ServicePlan, repository storage.Repository) ([]string, error) {
+	var platformTypes map[string]bool
```

Comment:
Why don't you initialize this like you do with the `platformNames`?

I see that's because you're differentiating whether the struct is not initialized (equal to nil) and when it's simply empty (not equal to nil, but len == 0) and executing some branching logic based on that.

But this makes the code inconsistent and hard to read. Please unify the variable initialization and instead make decisions based on if `len == 0`, rather than if `== nil`.

Reply:
refactored, please have a look at the new version